commands/FBClassDump.py

```diff
@@ -0,0 +1,190 @@
+    elif options.clsmethod:
+        printClassMethods(cls)
+    else:
+        printInstanceMethods(cls)
+
+def isClassObject(arg):
+    return runtimeHelpers.class_isMetaClass(runtimeHelpers.object_getClass(arg))
+
+def printInstanceMethods(cls, prefix='-'):
+    ocarray = instanceMethosOfClass(cls)
+    if not ocarray:
+      print "-- have none method or an error occur-- "
+      return
+
+    methodAddrs = covertOCArrayToPyArray(ocarray)
+    methods = []
+    for i in methodAddrs:
+      method = createMethodFromOCMethod(i)
+      if method is not None:
+        methods.append(method)
+        print prefix + ' ' + method.prettyPrint()
+
+def printClassMethods(cls):
+    printInstanceMethods(runtimeHelpers.object_getClass(cls), '+')
+
+# I find that a method that has variable parameter can not b.evaluateExpression
+# so I use numberWithLongLong: rather than -[NSString stringWithFormat:]
+def instanceMethosOfClass(klass):
+  tmpString = """
+    unsigned int outCount;
+    void **methods = (void **)class_copyMethodList((Class)$cls, &outCount);
+    NSMutableArray *result = [NSMutableArray array];
+    for (int i = 0; i < outCount; i++) {
+      NSNumber *num = (NSNumber *)[NSNumber numberWithLongLong:(long long)methods[i]];
+      [result addObject:num];
+    }
+    (void)free(methods);
+    id ret = result.count ? [result copy] : nil;
+    ret;
+  """
+
+  command = string.Template(tmpString).substitute(cls=klass)
+  command = '({' + command + '})'
+  ret = fb.evaluateExpressionValue(command)
+  if not ret.GetError().Success():
+    return None
+
+  ret = ret.GetValue()
+  if int(ret, 16) == 0: # return nil
+    ret = None
+  return ret
+
+# OC array only can hold id,
+# @return an array whose instance type is str of the oc object`s address
+
+def covertOCArrayToPyArray(oc_array):
+  is_array = fb.evaluateBooleanExpression("[{} isKindOfClass:[NSArray class]]".format(oc_array))
+  if not is_array:
+    return None
+
+  result = []
+  count = fb.evaluateExpression("(int)[{} count]".format(oc_array))
+
+  for i in range(int(count)):
+    value = fb.evaluateExpression("(id)[{} objectAtIndex:{}]".format(oc_array, i))
+    value = fb.evaluateExpression("(long long)[{} longLongValue]".format(value))
+    result.append(value)
+
+  return result
+
+
+class Method:
+
+  encodeMap = {
+    'c': 'char',
+    'i': 'int',
+    's': 'short',
+    'l': 'long',
+    'q': 'long long',
+
+    'C': 'unsigned char',
+    'I': 'unsigned int',
+    'S': 'unsigned short',
+    'L': 'unsigned long',
+    'Q': 'unsigned long long',
+
+    'f': 'float',
+    'd': 'double',
+    'B': 'bool',
+    'v': 'void',
+    '*': 'char *',
+    '@': 'id',
+    '#': 'Class',
+    ':': 'SEL',
+  }
+
+  def __init__(self, name, type_encoding, imp, oc_method):
+    self.name = name
+    self.type = type_encoding
+    self.imp = imp
+    self.oc_method = self.toHex(oc_method)
+
+  def prettyPrint(self):
+    # mast be bigger then 2, 0-idx for self, 1-st for SEL
```

Comment:
s/mast/must

I don't quite understand this comment, especially the "1-st" part.